### DISCUSSION TITLE
Using energy units as metadata for default variables may limit applicability to non-energy sectors

### DISCUSSION THREAD

**Discussion by FLomb** [calliope-project/calliope#511] [Custom constraints and math]
## Problem description

Model variables in the `math/base.yaml` file have been assigned `units` to populate the metadata. These units are currently energy specific, such as: `energy` or `power`. Nonetheless, Calliope is generally applicable, and has been applied in the past, also to non-energy flows and amounts, for instance, CO2 or water flows and stored amounts. Energy-specific units would conflict with this liberty.

One solution to have similar units but not energy specific might be to modify (i) `power` into `rate` and (ii) `energy` into `amount`, or something along these lines.

## Calliope version

0.6.11-dev

**Comment by brynpickering** [comment]
Possibilities for units from offline discussions:
- `energy_cap`: `carrier_unit_per_hour`
- `carrier_prod`/`con`: `carrier_unit`
- `storage`: `carrier_unit`

This is very generic but tries to deal with the fact that each `carrier` can have different units (e.g., energy: kWh, water: m3, ...), but does lose the info on the scale of things, i.e. all carriers should be of the same scale. Otherwise the system energy balance will be wrong.

**Comment by brynpickering** [comment]
Moving out of beta and int oa discussion thread since it is not critical to anything in particular and could do with more dicussing with a wider audience.
